zproject/computed_settings.py

```diff
@@ -1244,6 +1244,31 @@ def ensure_dict_path(d: dict[str, Any], keys: list[str]) -> None:
 EMAIL_GATEWAY_PASSWORD = get_secret("email_gateway_password")
 AUTH_LDAP_BIND_PASSWORD = get_secret("auth_ldap_bind_password", "")
 
+########################################################################
+# LiteLLM SETTINGS
+########################################################################
+
+# The model name that will used by LiteLLM library to configure
+# parameters to be sent to API.
+# The Llama-3-8B-instruct model is free to use and only requires submitting
+# a small form on the HuggingFace page for the model to gain access.
+# We only support HuggingFace and AWS Bedrock for LLM API requests.
+DEFAULT_TOPIC_SUMMARIZATION_MODEL: str | None = "huggingface/meta-llama/Meta-Llama-3-8B-Instruct"
+if PRODUCTION:
+    DEFAULT_TOPIC_SUMMARIZATION_MODEL = None
+TOPIC_SUMMARIZATION_MODEL = get_secret(
+    "topic_summarization_model", DEFAULT_TOPIC_SUMMARIZATION_MODEL
+)
+
+# Which API key to use will be determined based on TOPIC_SUMMARIZATION_MODEL.
+# HuggingFace access credentials
+HUGGINGFACE_API_KEY = get_secret("huggingface_api_key", None)
+
+# AWS Bedrock access credentials
+AWS_ACCESS_KEY_ID = get_secret("aws_access_key_id", None)
+AWS_SECRET_ACCESS_KEY = get_secret("aws_secret_access_key", None)
+AWS_REGION_NAME = get_secret("aws_region_name", None)
```

Comment:
So the way you've written these settings, one can only use Bedrock or HuggingFace as one's LiteLLM provider. This is not desirable -- end users should be able to just use whatever LiteLLM supports if we can make that work. 

I would expect that we could just have the ConfigParser read a `litellm` section of `zulip-secrets.conf` and store those key/value pairs into an object that is directly formatted into `litellm_params` above, such that our documentation could be just something like:

> In /etc/zulip/zulip-secrets.conf, create a `[litellm]` section, and within that section, enter the LiteLLM parameters expected as the `litellm_params` for use of that model. 

And then we can have examples in the docs for specific providers.

Reply:
I think we can merge with this version and discuss in #backend how best to proceed. Maybe it'll make sense for @alexmv to do the PR reworking this configuration detail. We also need to figure out where we're documenting this in the production docs.

Reply:
Okay, happy to get help from Alex on this.